lib/plsql/procedure.rb

```diff
@@ -350,18 +371,35 @@ def get_field_definitions(argument_metadata) #:nodoc:
       case argument_metadata[:type_object_type]
       when "PACKAGE"
         @schema.select_all(
-          "SELECT attr_no, attr_name, attr_type_owner, attr_type_name, attr_type_package, length, precision, scale, char_used
-          FROM ALL_PLSQL_TYPES t, ALL_PLSQL_TYPE_ATTRS ta
-          WHERE t.OWNER = :owner AND t.type_name = :type_name AND t.package_name = :package_name
-          AND ta.OWNER = t.owner AND ta.TYPE_NAME = t.TYPE_NAME AND ta.PACKAGE_NAME = t.PACKAGE_NAME
-          ORDER BY attr_no",
-          @schema_name, argument_metadata[:type_name], argument_metadata[:type_subname]) do |r|
+          "SELECT ta.attr_no, ta.attr_name,
+            nvl(s.table_owner, attr_type_owner) attr_type_owner,
+            CASE WHEN ta.attr_type_package IS NOT NULL THEN ta.attr_type_name ELSE nvl(s.table_name, ta.attr_type_name) END attr_type_name,
+            CASE WHEN ta.attr_type_package IS NOT NULL THEN nvl(s.table_name, ta.attr_type_package) END attr_type_package,
```

Comment:
@joschug 
Just wanted to check if you intend to write 
`CASE WHEN ta.attr_type_package IS NOT NULL THEN **ta.attr_type_package ELSE** nvl(s.table_name, ta.attr_type_package) END attr_type_package,`

and end up missing the ELSE part.